### ISSUE TITLE
Distribute the sequence diagrams to the team members (one - two diagrams per member as suggested by the TA).

### ISSUE THREAD

**Issue by ebrarkiziloglu** [bounswe/bounswe2024group5#111]

**Comment by ebrarkiziloglu** [comment]
The deadline of the sequence diagrams is updated to be April 28th.

**Comment by HutkuC** [comment]
Wiki page for sequence diagrams:
[https://github.com/bounswe/bounswe2024group5/wiki/Sequence-Diagrams](https://github.com/bounswe/bounswe2024group5/wiki/Sequence-Diagrams)

**Comment by set120120** [comment]
I've added "Filter User Profiles" and "Filter Songs". Can someone give feedback to me?

**Comment by semihYILMAZ37** [comment]
> I've added "Filter User Profiles" and "Filter Songs". Can someone give feedback to me?
@set120120 

It could be better to include Filter Object. User can pass message "initiate(List~Key, Value~)" to Filter Class. After initiating a filter object, it should be passed to search engine class with search{Users/Posts}( filter: Filter) method.

**Comment by semihYILMAZ37** [comment]
I have added change profile details diagram and open to feedback from anyone.

**Comment by set120120** [comment]
> > I've added "Filter User Profiles" and "Filter Songs". Can someone give feedback to me?
> > @set120120
> 
> It could be better to include Filter Object. User can pass message "initiate(List<Key, Value>)" to Filter Class. After initiating a filter object, it should be passed to search engine class with search{Users/Posts}( filter: Filter) method.

Thank you for ideas. I revised filter songs/userProfiles sections based on feedback.

**Comment by set120120** [comment]
Instead of using "updateLikes(post_id, updated_likes)" in the 9th and 10th diagrams, you could express it more descriptively for regular user. For instance, you could say "the post has been liked" or "a comment has been submitted." @ebrarkiziloglu

**Comment by ebrarkiziloglu** [comment]
Thank you @set120120, I applied your feedback.

**Comment by set120120** [comment]
I reviewed all the diagrams. I made 3 small changes. Everything else looks okay.
Change 1 -> I connected the follow/unfollow diagram to a single sub.
Change 2 -> The largest child in the Create post diagram has been deleted.
Change 3 -> I wrote Map<K, V> instead of username: string, password: string in the Register diagram. Because more than 2 pieces of information are required when registering.
In short, all diagrams are ready. Good job, team! ❤️
